pkg/cmd/cli/cmd/login/loginoptions.go

```diff
@@ -121,7 +121,7 @@ func (o *LoginOptions) getClientConfig() (*restclient.Config, error) {
 		// certificate authority unknown, check or prompt if we want an insecure
 		// connection or if we already have a cluster stanza that tells us to
 		// connect to this particular server insecurely
-		case x509.UnknownAuthorityError:
+		case x509.UnknownAuthorityError, x509.HostnameError, x509.CertificateInvalidError:
 			if o.InsecureTLS ||
 				hasExistingInsecureCluster(*clientConfig, *o.StartingKubeConfig) ||
 				promptForInsecureTLS(o.Reader, o.Out) {
```

Comment:
promptForInsecureTLS needs to tweak the text based on the error... pass in the error, and switch inside promptForInsecureTLS on the error type